### PR TITLE
Add logic to track speed of receiving join ops and reset connection on timeout

### PR DESCRIPTION
**Pull request by vladsud**
Add logic to track speed of receiving join ops.
This is related to https://onedrive.visualstudio.com/SPIN/_queries/edit/1156694/?triage=true
Related PR: https://github.com/microsoft/FluidFramework/pull/6928/

### CODE REVIEW

File: packages/loader/container-loader/src/connectionStateHandler.ts [hunk 1]
```diff
@@ -65,11 +67,30 @@ export class ConnectionStateHandler extends EventEmitterWithErrorHandling<IConne
...
+        this.joinOpTimer = new Timer(
+            15000,
+            () => {
+                // If this event fires too often, then we should look into one of the options:
+                // 1. If it happens because it takes client too long to catch up, then we should consider why
+                //    and how to address it. It blocks collab window moving forward. One reason might be -
+                //    we raised "connected" event to runtime right away on establishing "read" connection, allowing
+                //    it to send op and switch to "write" mode too early. We should consider waiting with delivery
+                //    of initial "connected" signal until client is mostly caught (see waitContainerToCatchUp())
+                // 2. It's possible that client does not receiving any ops at all on this connection, either due
```
Comment (jatgarg) [inline packages/loader/container-loader/src/connectionStateHandler.ts:80]: nit: receiving -> receive

File: packages/loader/container-loader/src/connectionStateHandler.ts [hunk 2]
```diff
@@ -65,11 +67,30 @@ export class ConnectionStateHandler extends EventEmitterWithErrorHandling<IConne
                 this.applyForConnectedState("timeout");
             },
         );
+
+        this.joinOpTimer = new Timer(
+            15000,
+            () => {
+                // If this event fires too often, then we should look into one of the options:
+                // 1. If it happens because it takes client too long to catch up, then we should consider why
```
Comment (jatgarg) [inline packages/loader/container-loader/src/connectionStateHandler.ts:75]: nit: Remove "if" at starting.

File: packages/loader/container-loader/src/deltaManager.ts [hunk 3]
```diff
@@ -399,21 +399,34 @@ export class DeltaManager
...
+                    // If we have pending changes, then we will never send them - it smells like
+                    // host logic error.
+                    this.logger.sendErrorEvent({ eventName: "ForceReadonlyPendingChanged" });
+                }
+
+                this.disconnectFromDeltaStream("Force readonly");
             }
             safeRaiseEvent(this, this.logger, "readonly", this.readonly);
-            if (reconnect) {
+            if (this.reconnectMode === ReconnectMode.Enabled) {
```
Comment (jatgarg) [inline packages/loader/container-loader/src/deltaManager.ts:416]: Aren't we digressing from the logic here? Previously we used to disconnect and check if we need to reconnect and then triggered connection. Now we are disconnectFromDeltaStream if this.readOnly is true but below we are just reconnecting based on ReconnectMode even if we didn't disconnected first? Is it ok to do so?

Comment (vladsud) [reply packages/loader/container-loader/src/deltaManager.ts:416]: Yep, it's wrong :) I'll fix it - thanks for catching it!

**Review by jatgarg — APPROVED** [review]

**Pull request merged**